src/spatial/flatten.jl

```diff
@@ -0,0 +1,73 @@
+using DiffEqJump, DiffEqBase, LightGraphs
+
+rates = [0.1, 1.0]
+netstoch = [[1 => -1, 2 => -1, 3 => 1], [1 => 1, 2 => 1, 3 => -1]]
+reactstoch = [[1 => 1, 2 => 1], [3 => 1]]
+
+graph = grid((2,2))
+num_species = 3
+num_nodes = nv(graph)
+end_time = 1.0
+tspan = (0.0, end_time)
+u0 = ones(Int, num_species, num_nodes)
+alg = Direct()
+
+hopping_rate = 1.0
+hopping_constants = Vector{Matrix{Float64}}(undef, num_nodes) # TODO QUESTION is this how we want to accept hopping constants? 
+for site in 1:num_nodes
+    hopping_constants[site] = hopping_rate*ones(num_species, DiffEqJump.num_neighbors(graph, site))
+end
+
+#TODO QUESTION should accept variable rx rates?
+
+# reaction
+function flatten(netstoch::AbstractArray, reactstoch::AbstractArray, rates::AbstractArray, graph::AbstractGraph, u0::Matrix{Int}, tspan, alg, hopping_constants::Vector{Matrix{F}}; kwargs...) where F <: Number
```

Comment:
The reason I want to take `rates` as input is that in `MassActionJump` only scaled rates are stored. I could write a function called `unscalerate` -- inverse to `scalerate`. Then I would first unscale rates from massaction jumps and then flatten. If you think it would be better, I'll go ahead and do that.

Reply:
Why would you need to unscale the rate? The scalings there should still be applied in the spatial case. They are scalings due to the reaction order, not volumes or such. It isn't generally true that `p` maps directly to the rates in a `MassActionJump`, so you need to work with `scaled_rates` as the rate constants.

Reply:
I added a keyword arg `scale_rates`, which is false in case the passed in rates come from `MassActionJump`